Mathlib/CategoryTheory/Bicategory/Modification/Oplax.lean

```diff
@@ -122,15 +132,137 @@ lemma Modification.comp_app' {X : B} {F G : B ⥤ᵒᵖᴸ C} {α β γ : F ⟶
+  app a := Γ.app a
+
+instance hasCoeToOplax :
+    Coe (Modification η θ) (OplaxTrans.Modification η.toOplax θ.toOplax) :=
+  ⟨toOplax⟩
+
+/-- The modification between strong transformations of oplax functors associated to a modification
+between the underlying oplax transformations. -/
+@[simps]
+def mkOfOplax (Γ : OplaxTrans.Modification η.toOplax θ.toOplax) : Modification η θ where
+  app a := Γ.app a
+  naturality f := by simpa using Γ.naturality f
+
+/-- Modifications between strong transformations of oplax functors are equivalent to modifications
+between the underlying oplax transformations. -/
+@[simps]
+def equivOplax : (OplaxTrans.Modification η.toOplax θ.toOplax) ≃ Modification η θ where
+  toFun := mkOfOplax
+  invFun := toOplax
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+section
+
+variable (Γ : Modification η θ) {a b c : B} {a' : C}
+
+@[reassoc (attr := simp)]
+theorem whiskerLeft_naturality (f : a' ⟶ F.obj b) (g : b ⟶ c) :
+    f ◁ F.map g ◁ Γ.app c ≫ f ◁ (θ.naturality g).hom =
+      f ◁ (η.naturality g).hom ≫ f ◁ Γ.app b ▷ G.map g :=
+  OplaxTrans.Modification.whiskerLeft_naturality Γ.toOplax _ _
+
+@[reassoc (attr := simp)]
+theorem whiskerRight_naturality (f : a ⟶ b) (g : G.obj b ⟶ a') :
+    F.map f ◁ Γ.app b ▷ g ≫ (α_ _ _ _).inv ≫ (θ.naturality f).hom ▷ g =
+      (α_ _ _ _).inv ≫ (η.naturality f).hom ▷ g ≫ Γ.app a ▷ G.map f ▷ g :=
+  OplaxTrans.Modification.whiskerRight_naturality Γ.toOplax _ _
+
+end
+
+variable (η) in
+/-- The identity modification. -/
+@[simps]
+def id : Modification η η where app a := 𝟙 (η.app a)
+
+instance : Inhabited (Modification η η) :=
+  ⟨Modification.id η⟩
+
+/-- Vertical composition of modifications. -/
+@[simps]
+def vcomp {ι : F ⟶ G} (Γ : Modification η θ) (Δ : Modification θ ι) : Modification η ι where
+  app a := Γ.app a ≫ Δ.app a
+
+end Modification
+
+/-- Category structure on the strong natural transformations between oplax functors. -/
+@[simps!]
+scoped instance homCategory : Category (F ⟶ G) where
+  Hom := Modification
+  id := Modification.id
+  comp := Modification.vcomp
+
+instance : Inhabited (Modification η η) :=
+  ⟨𝟙 η⟩
+
+@[ext]
+lemma homCategory.ext {m n : η ⟶ θ} (w : ∀ b, m.app b = n.app b) : m = n :=
+  Modification.ext (funext w)
+
+/-- Construct a modification isomorphism between strong natural transformations (of oplax functors)
+by giving object level isomorphisms, and checking naturality only in the forward direction.
+-/
+@[simps]
+def isoMk (app : ∀ a, η.app a ≅ θ.app a)
+    (naturality :
+      ∀ {a b} (f : a ⟶ b),
+        F.map f ◁ (app b).hom ≫ (θ.naturality f).hom =
+          (η.naturality f).hom ≫ (app a).hom ▷ G.map f := by cat_disch) : η ≅ θ where
+  hom := { app a := (app a).hom }
+  inv :=
+    { app a := (app a).inv
+      naturality {a b} f := by
+        simpa using _ ◁ (app b).inv ≫= (naturality f).symm =≫ (app a).inv ▷ _ }
+
+@[deprecated (since := "2025-11-11")] alias ModificationIso.ofComponents := isoMk
+
```

Comment:
```suggestion
```
Since `Oplax.StrongTrans.ModificationIso.ofComponents` was not a thing, no need to deprecate it I guess.